src/org/thoughtcrime/securesms/connect/ApplicationDcContext.java

```diff
@@ -397,6 +403,27 @@ public void waitForThreadsExecutedOnce() {
     }
   }
 
+  public void stopThreads() {
+    Log.i(TAG, "!!!!!!!!!!!! Stopping threads ...");
+    run = false;
+    synchronized (threadsCritical) {
+      if (imapThread!=null) { interruptImapIdle(); }
+      if (mvboxThread!=null) { interruptMvboxIdle(); }
+      if (sentboxThread!=null) { interruptSentboxIdle(); }
+      if (smtpThread!=null) { interruptSmtpIdle(); }
+      while (true) {
+        if ( (imapThread==null || !imapThread.isAlive())
+          && (mvboxThread==null || !mvboxThread.isAlive())
+          && (sentboxThread==null || !sentboxThread.isAlive())
+          && (smtpThread==null || !smtpThread.isAlive())) {
+          break;
+        }
+        Util.sleep(100);
```

Comment:
It is probably better to keep interrupting all alive threads in this loop until they stop. Interrupt may not work the first time due to https://github.com/deltachat/deltachat-core-rust/issues/925.

Reply:
good hint, great! i do a commit targeting that. i also have one other issue found ...